File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(),nums.end());
        int ml = 0;
        for(auto i : nums){
            if(s.count(i)){
                int t= 0 , j = i ;
                while(s.count(++j)){
                    t++;
                    s.erase(j);
                }
                j = i;
                while(s.count(--j)){
                    s.erase(j);
                    t++;
                }
                ml = max(ml,t+1);
            }
        }
        return ml;
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-0.java
class Solution {

        public boolean isValid(int[] weights, int days, int m) {
        int t = m, count = 1, sum = 0;
        for (int weight : weights) {
            // System.out.println(sum + weight);
            if (sum + weight <= t) {
                sum += weight;
            } else {
                if (weight > t)
                    return false;
                sum = weight;
                count++;
                if (count > days)
                    return false;
            }
        }
        // System.out.println("m : "+ m + "count : " + count);
        return count <= days;
    }

    public int shipWithinDays(int[] weights, int days) {
        int l = Arrays.stream(weights).max().getAsInt();
        ;
        int h = Arrays.stream(weights).sum();
        // System.out.println(h);
        int ans = h;
        while (l < h) {
            int m = l + (h - l) / 2;
            // System.out.println(m);
            if (isValid(weights, days, m)) {
                ans = m;
                h = m;
            } else {
                l = m + 1;
            }
        }
        return ans;
    }

}